django/utils/dateparse.py

```diff
@@ -87,12 +90,15 @@ def parse_time(value):
     Return None if the input isn't well formatted, in particular if it
     contains an offset.
     """
-    match = time_re.match(value)
-    if match:
-        kw = match.groupdict()
-        kw['microsecond'] = kw['microsecond'] and kw['microsecond'].ljust(6, '0')
-        kw = {k: int(v) for k, v in kw.items() if v is not None}
-        return datetime.time(**kw)
+    try:
+        return datetime.time.fromisoformat(value)
+    except (TypeError, ValueError):
```

Comment:
IMO we don't need to catch `TypeError` because it will only cause another `TypeError` in the `.match()` call, so I would prefer
```
    return datetime.date.fromisoformat(value)
TypeError: fromisoformat: argument must be str
```
instead of
```
    match = date_re.match(value)
TypeError: expected string or bytes-like object
```